src/test/kotlin/org/rust/ide/inspections/RsDeprecationInspectionTest.kt

```diff
@@ -0,0 +1,255 @@
+            One,
+            Two,
+            Three
+        }
+
+        fn main() {
+            <warning descr="`Numbers` is deprecated">Numbers</warning>::One;
+        }
+    """)
+
+    fun `test deprecated enum item`() = checkByText("""
+        enum Numbers {
+            #[deprecated]
+            One,
+            Two,
+            Three
+        }
+
+        fn main() {
+            Numbers::<warning descr="`One` is deprecated">One</warning>;
+        }
+    """)
+
+    fun `test deprecated type alias`() = checkByText("""
+        #[deprecated]
+        type Name = String;
+
+        fn main() {
+            let name: <warning descr="`Name` is deprecated">Name</warning> = "test".to_string();
+        }
+    """)
+
+    fun `test deprecated variable`() = checkByText("""
+        #[deprecated]
+        static SOME_INT: i32 = 5;
+
+        fn main() {
+            let a = <warning descr="`SOME_INT` is deprecated">SOME_INT</warning>;
+        }
+    """)
+
+    fun `test deprecated inline module`() = checkByText("""
+        #[deprecated]
+        mod foo {
+            pub static SOME_INT: i32 = 5;
+        }
+
+        fn main() {
+            let a = <warning descr="`foo` is deprecated">foo</warning>::SOME_INT;
+        }
+    """)
+
+    fun `test deprecated variable in struct`() = checkByText("""
+        struct Foo {
+            #[deprecated]
+            pub x: i32
+        }
+
+        fn main() {
+            let a = Foo { <warning descr="`x` is deprecated">x</warning>: 123 };
+        }
+    """)
+
+    fun `test deprecated non-inline module`() = checkByFileTree("""
+        //- main.rs
+            #[deprecated]
+            mod foo;
+
+            fn main() {
+                let a = /*caret*/<warning descr="`foo` is deprecated">foo</warning>::SOME_INT;
+            }
+        //- foo.rs
+            pub static SOME_INT: i32 = 5;
+    """)
+
+    fun `test allow deprecated function`() = checkByText("""
+        #[deprecated]
+        pub fn foo() {
+        }
+
+        #[allow(deprecated)]
+        fn main() {
+            foo();
+        }
+    """)
+
+    fun `test allow deprecated statement`() = checkByText("""
+        #[deprecated]
+        pub fn foo() {
+        }
+
+        fn main() {
+            #[allow(deprecated)]
+            foo();
+        }
+    """)
+
+    fun `test allow deprecated item in module`() = checkByText("""
+        #[allow(deprecated)]
+        mod foo {
+            #[deprecated]
+            static SOME_INT: i32 = 5;
+            static OTHER_INT: i32 = SOME_INT;
+        }
+    """)
+
+    fun `test allow deprecated inner attribute`() = checkByText("""
+        #![allow(deprecated)]
+
+        #[deprecated]
+        fn foo() {
+            println!("TEST")
+        }
+
+        fn main() {
+            foo();
+        }
+    """)
+
+    fun `test AST is not loaded when deprecated item in another file`() {
+        checkAstNotLoaded(VirtualFileFilter { "foo.rs" == it.name })
```

Comment:
Unfortunately, it doesn't work for highlighting checking (including inspection tests) because inside such tests `VirtualFileFilter` is changed by test code itself. So at this moment, it tests nothing
Let's replace it with temporary hack
```suggestion
        (myFixture as CodeInsightTestFixtureImpl).setVirtualFileFilter { !it.path.endsWith("main.rs") }
```